index.d.ts

```diff
@@ -0,0 +1,60 @@
+import {MergeExclusive} from 'type-fest';
+
+type Options = MergeExclusive<
+	{
+		/**
+		_You usually won't need this option. Specify it only when actually needed._
+
+		File extension. Mutually exclusive with the `name` option.
+		*/
+		readonly extension?: string;
+	},
+	{
+		/**
+		_You usually won't need this option. Specify it only when actually needed._
+
+		Filename. Mutually exclusive with the `extension` option.
+		*/
+		readonly name?: string;
+	}
+>;
+
+/**
+Get a temporary file path you can write to.
+
+@example
+```
+import * as tempy from tempy';
```

Comment:
I don't really want to recommend using named imports here. I don't intend to expose them as such when eventually moving to ESM. See: https://github.com/sindresorhus/meta/issues/6

Reply:
Not sure what you propose instead. Can you give me a hint?

Reply:
`import tempy = require('tempy');` for now and `import tempy from 'tempy';` later on when using ESM.

Reply:
I can change this but you have to know that nothing will forbid users to use the `import * as tempy` syntax even without `esModulesInterop`.

Reply:
I’m aware, but at least I’ve shown the recommended way.